services/httpd/handler.go

```diff
@@ -1165,9 +1165,30 @@ func cors(inner http.Handler) http.Handler {
 
 func requestID(inner http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		uid := uuid.TimeUUID()
-		r.Header.Set("Request-Id", uid.String())
-		w.Header().Set("Request-Id", r.Header.Get("Request-Id"))
+		// X-Request-ID takes priority.
+		var rid = r.Header.Get("X-Request-ID")
```

Comment:
I learned through previous benchmarking that we can avoid some garbage creation by following canonicalization rules in our calls to Header.Get. This should be `-Id`, not `-ID`.